docs/dev/upgrade-opentelemetry.md

```diff
@@ -2,12 +2,27 @@
 
 When the [opentelemetry-collector](https://github.com/open-telemetry/opentelemetry-collector) and [opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib) repos do a release, we need to update our dependencies to pick up the new changes.
 
+## Update All Dependencies
```

Comment:
I collapsed the update, install, and generate into a single make target. Since on most OpenTelemetry upgrades build and test fail by default anyway, I decided to leave those targets separate since they will likely need lots of manual work anyway to get each one to pass.